### PR TITLE
Code example for Pubsub component using Dapr Java SDK

### PR DESCRIPTION
**Pull request by pravinpushkar**
Signed-off-by: Pravin Pushkar <ppushkar@microsoft.com>

# Description

Code example for pubsub component using Dapr java SDK

## Issue reference

We strive to have all PRs being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: NA

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] The quickstart code compiles correctly
* [x] You've tested new builds of the quickstart if you changed quickstart code
* [x] You've updated the quickstart's README if necessary
* [ ] If you have changed the steps for a quickstart be sure that you have updated the automated validation accordingly. All of our quickstarts have annotations that allow them to be executed automatically as code. For more information see [mechanical-markdown](https://github.com/dapr/mechanical-markdown#readme). For user guide with examples see [Examples](https://github.com/dapr/mechanical-markdown/tree/main/examples#mechanical-markdown-by-example).

### CODE REVIEW

File: pub_sub/java/sdk/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Dapr pub/sub
+
+In this quickstart, you'll create a publisher microservice and a subscriber microservice to demonstrate how Dapr enables a publish-subcribe pattern. The publisher will generate messages of a specific topic, while subscribers will listen for messages of specific topics. See [Why Pub-Sub](#why-pub-sub) to understand when this pattern might be a good choice for your software architecture.
```
Comment (mukundansundar) [inline pub_sub/java/sdk/README.md:3]:
```suggestion
In this quickstart, you'll create a publisher microservice and a subscriber microservice that demonstrates how Dapr enables a publish-subscribe pattern. The publisher will publish messages to a specific topic, while subscribers will listen for messages from a specific topic. See [Why Pub-Sub](#why-pub-sub) to understand when this pattern might be a good choice for your software architecture.
```

File: pub_sub/java/sdk/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+- Java client message generator `checkout`
+
+And one subscriber:
+
+- Java subscriber `order-processor`
+
+## Pre-requisites
+
+* [Dapr and Dapr Cli](https://docs.dapr.io/getting-started/install-dapr/).
+* Java JDK 11 (or greater): [Oracle JDK](https://www.oracle.com/technetwork/java/javase/downloads/index.html#JDK11) or [OpenJDK](https://jdk.java.net/13/).
```
Comment (mukundansundar) [inline pub_sub/java/sdk/README.md:20]: provide the same version for openjdk

File: pub_sub/java/sdk/README.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+* [Apache Maven](https://maven.apache.org/install.html) version 3.x.
+### Run Java message publisher app with Dapr
```
Comment (mukundansundar) [inline pub_sub/java/sdk/README.md:22]: blank line above.

File: pub_sub/java/sdk/order-processor/src/main/java/com/service/controller/OrderProcessingServiceController.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    private static final Logger log = LoggerFactory.getLogger(OrderProcessingServiceController.class);
+
+    @Topic(name = "orders", pubsubName = "order_pub_sub")
+    @PostMapping(path = "/orders")
+    public Mono<ResponseEntity> getCheckout(@RequestBody(required = false) CloudEvent<String> cloudEvent) {
+        return Mono.fromSupplier(() -> {
+            try {
+                log.info("Subscriber received: " + cloudEvent.getData());
+                return new ResponseEntity<>("successful",
```
Comment (mukundansundar) [inline pub_sub/java/sdk/order-processor/src/main/java/com/service/controller/OrderProcessingServiceController.java:27]:
```suggestion
                return new ResponseEntity<>("SUCCESS",
```

File: pub_sub/java/sdk/order-processor/src/main/java/com/service/controller/OrderProcessingServiceController.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import org.slf4j.LoggerFactory;
+
+import reactor.core.publisher.Mono;
+
+@RestController
+public class OrderProcessingServiceController {
+
+    private static final Logger log = LoggerFactory.getLogger(OrderProcessingServiceController.class);
+
+    @Topic(name = "orders", pubsubName = "order_pub_sub")
```
Comment (paulyuk) [inline pub_sub/java/sdk/order-processor/src/main/java/com/service/controller/OrderProcessingServiceController.java:21]: Nice!

File: pub_sub/java/sdk/order-processor/src/main/java/com/service/controller/OrderProcessingServiceController.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    @PostMapping(path = "/orders")
+    public Mono<ResponseEntity> getCheckout(@RequestBody(required = false) CloudEvent<String> cloudEvent) {
+        return Mono.fromSupplier(() -> {
+            try {
+                log.info("Subscriber received: " + cloudEvent.getData());
+                return new ResponseEntity<>("successful",
+                        HttpStatus.OK);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
```
Comment (paulyuk) [inline pub_sub/java/sdk/order-processor/src/main/java/com/service/controller/OrderProcessingServiceController.java:30]: Is it normal to show try catch with a rethrow?  or just cut the try catch?

File: pub_sub/java/sdk/checkout/src/main/java/com/service/CheckoutServiceApplication.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+import io.dapr.client.DaprClient;
+import io.dapr.client.DaprClientBuilder;
+import io.dapr.client.domain.Metadata;
+
+import static java.util.Collections.singletonMap;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+public class CheckoutServiceApplication {
+	public static void main(String[] args) throws InterruptedException{
```
Comment (paulyuk) [inline pub_sub/java/sdk/checkout/src/main/java/com/service/CheckoutServiceApplication.java:12]: I like it that you have this in a single sub main vs. refactoring into several functions (which I've seen in other braches).  Please continue with this level of simplicity.

**Review by paulyuk — CHANGES_REQUESTED** [review]
Please see a few of the comments and suggestions.

File: pub_sub/java/sdk/order-processor/src/main/java/com/service/controller/OrderProcessingServiceController.java (hunk 6 above)
Comment (pravinpushkar) [reply pub_sub/java/sdk/order-processor/src/main/java/com/service/controller/OrderProcessingServiceController.java:30]: I think yes, it is fine to write and here I wanted to throw if there is any unchecked exception happened like the `cloudEvent.getData` comes out to be null.

File: pub_sub/java/sdk/README.md (hunk 1 above)
Comment (pravinpushkar) [reply pub_sub/java/sdk/README.md:3]: We have similar wordings at other examples, So to keep it consistent I have used similar wordings.

**Review by paulyuk — APPROVED** [review]
LGTM.  thank you for addressing the feedback

**Pull request closed without merging**